hathor/p2p/manager.py

```diff
@@ -107,6 +110,30 @@ def has_synced_peer(self) -> bool:
                 return True
         return False
 
+    def is_practically_synced(self) -> bool:
+        """ Returns whether we can consider for practical resons that we are synced to the network.
+
+        This method name is intentionally a bit vague to give room for some imprecisions and subjective protection
+        against malicious or rogue nodes. This method is primarily intended for use by APIs that need to wait for the
+        sync to finish but can work when it hasn't (even with bad, but not catastrophic consequences).
+
+        For example: `get_block_template` API can work at anytime, but before the node has synced, it will lead to
+        blocks way into the past with too low difficulty (because of decay), which is bad, but not catastrophic.
+        """
+        # How it works: filter out the peers that seem to have a reasonable timestamp (not way into the future, and not
+        # way into the past). Test if we're synced with any of them.
+        timestamp = int(self.reactor.seconds())
+        sync_threshold = settings.PRACTICALLY_SYNC_THRESHOLD
+        reasonable_timestamp_range = range(timestamp - sync_threshold, timestamp + sync_threshold)
+        for conn in self.get_ready_connections():
+            assert conn.state is not None
+            assert isinstance(conn.state, ReadyState)
+            if conn.state.get_sync_plugin().synced_timestamp not in reasonable_timestamp_range:
```

Comment:
I think we should use an interval comparison instead of a list. We can do it in O(1) using two comparisons instead of in O(n) using a list.

Reply:
I'm not using a list, `range` is its own type that is efficient for this.

Reply:
I didn't know the `range` type. I just read Python's documentation. Thanks!